webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/domain/reports/ExpectedReports.scala

```diff
@@ -199,350 +204,593 @@ final case class NodeConfigIdInfo(
-      (
-          (json \ "componentName" )
-        , (json \ "values")
-        , (json \ "values").extractOpt[List[String]]
-        , (json \ "unexpanded").extractOpt[List[String]]
-        , (json \ "subComponents") match {
-        case JArray(subs) => Some(sequence(subs)(component))
-        case _=> None
+    final case class JsonDirectiveExpecteReports7_1(
+        did: DirectiveId
+      , pm : Option[PolicyMode]
+      , s  : Option[Boolean]
+      , cs : List[JsonComponentExpectedReport7_1]
+    ) {
+      def transform = DirectiveExpectedReports(did, pm, s.getOrElse(false), cs.map(_.transform))
+    }
+    implicit class _JsonDirectiveExpecteReports7_1(x: DirectiveExpectedReports) {
+      def transform = {
+        val s = if(x.isSystem) Some(true) else None
+        JsonDirectiveExpecteReports7_1(
+          x.directiveId, x.policyMode, s, x.components.map(_.transform)
+        )
       }
+    }
+
+    final case class JsonRuleExpectedReports7_1(
+        rid: RuleId
+      , ds : List[JsonDirectiveExpecteReports7_1]
+    ) {
+      def transform = RuleExpectedReports(rid, ds.map(_.transform))
+    }
+    implicit class _JsonRuleExpectedReports7_1(x: RuleExpectedReports) {
+      def transform = JsonRuleExpectedReports7_1(x.ruleId, x.directives.map(_.transform))
+    }
 
-        , (json \ "reportingLogic").extractOpt[String]
-     ) match {
-        case (JString(name), values, None, _, None, None ) =>
-              for {
-                values <- values match {
-                  case JArray(arrayValues) => sequence(arrayValues)(value)
-                  case _ => Failure("")
-                }
-              } yield {
-                ValueExpectedReport(name, values.toList)
-
-              }
-        case (JString(name), _, Some(values), Some(unexpanded), None, None ) =>
-          Full(ValueExpectedReport(name, values.zip(unexpanded).map(ExpectedValueMatch.tupled)))
-        case (JString(name),_,  _, _, Some(Full(sub)), Some(composition) )=>
-          for {
-            reportingLogic <-  ReportingLogic.parse(composition).toBox
-          } yield {
-            BlockExpectedReport(name, reportingLogic, sub.toList)
+    final case class JsonNodeExpectedReports7_1(
+        ms: JsonModes7_1
+      , rs: List[JsonRuleExpectedReports7_1]
+      , os: List[JsonOverrides7_1]
+    ) extends JsonNodeExpectedReportV {
+      def transform = JsonNodeExpectedReports(ms.transform, rs.map(_.transform), os.map(_.transform))
+    }
+    implicit class _JsonNodeExpecteReports7_1(x: JsonNodeExpectedReports) {
+      def transform = JsonNodeExpectedReports7_1(x.modes.transform, x.ruleExpectedReports.map(_.transform), x.overrides.map(_.transform))
+    }
+
+
+
+    ////////// json codec //////////
+
+    implicit val codecJsonGlobalPolicyMode7_1: JsonCodec[JsonGlobalPolicyMode7_1] = DeriveJsonCodec.gen
+    implicit val codecJsonAgentRun7_0: JsonCodec[JsonAgentRun7_1] = DeriveJsonCodec.gen
```

Comment:
is it a typo ?

Reply:
yep. The name does not matter, but it's quite misleading :)